File: MaxLength_Word_String/Max.py
"""
    Author : Pakshal Shashikant Jain
    Date : 23/09/2023
    Program : Write a program which accept string from user and return length of
              largest word.

              Input : "Marvellous Multi OS Infosystems"
              Output : 11
"""

###############################################################################################################################################################################

#Function To Find Out Length of Largest Word
def MaxWordLength(arr) :
    i = 0;
    icnt = 0;
    iMax = 0;

    if(arr == None) :
        return;

    #Logic To Find
    while(i != len(arr)) :
        if(arr[i] == ' ') :
            while((arr[i] == ' ')and(i != len(arr))) :
                i = i + 1;
            icnt = 0;
        else :
            icnt = icnt + 1;
            if(icnt > iMax) :
                iMax = icnt;
            i = i + 1;
    
    return iMax;

#######################################################################################################################################################################

#Main Function
def main() :
    iret = 0;
    str = " ";
    arr = [];

    print("Jay Ganesh....");

    #Taking Input from user
    print("Enter One String");
    str = input();

    #Creating Character array of Entered String
    arr = list(str);

    #Call To TO MaxWordLength Function 
    iret = MaxWordLength(arr);

    #Printing Length of Largest Word
    print("Length of Largest Word is : ",iret);

########################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: LargestDigitInNum_RecursiveApproach/Max.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/5/2024
    Program : Write a recursive program which accept number from user and return
              largest digit

              Input : 87983
              Output : 9   
*/

//Required Libraries
package main
import "fmt"

var imax int = 0;         //Global Variable
 
//Function To Calculate Largest Digit
func Max(ino int) int {
	var Digit int = 0;

	//Logic To Find
	Digit = ino % 10;
	if(imax < Digit) {
		imax = Digit;
	}
	ino = ino / 10;

	if(ino != 0) {
		//Recursive Function Call To Max Function
		Max(ino);
	} 

	return imax;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var ino int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");

	//Taking Input From User
	fmt.Println("Enter One Number : ");
	fmt.Scan(&ino);

	//Call To Max Function
	iret = Max(ino);
	
	//Printing Largest Digit
	fmt.Printf("Largest Digit in Entered Number %d is : %d",ino,iret);
}